cmd/common.go

```diff
@@ -0,0 +1,146 @@
+	c "github.com/centrifuge/go-centrifuge/context"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/keytools"
+	"github.com/centrifuge/go-centrifuge/node"
+	"github.com/centrifuge/go-centrifuge/queue"
+	"github.com/centrifuge/go-centrifuge/storage"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+var log = logging.Logger("centrifuge-cmd")
+
+func createIdentity(idService identity.Service) (identity.CentID, error) {
+	centID := identity.RandomCentID()
+	_, confirmations, err := idService.CreateIdentity(centID)
+	if err != nil {
+		return [identity.CentIDLength]byte{}, err
+	}
+	_ = <-confirmations
+
+	return centID, nil
+}
+
+func generateKeys(config config.Config) {
+	p2pPub, p2pPvt := config.GetSigningKeyPair()
+	ethAuthPub, ethAuthPvt := config.GetEthAuthKeyPair()
+	keytools.GenerateSigningKeyPair(p2pPub, p2pPvt, "ed25519")
+	keytools.GenerateSigningKeyPair(p2pPub, p2pPvt, "ed25519")
+	keytools.GenerateSigningKeyPair(ethAuthPub, ethAuthPvt, "secp256k1")
+}
+
+func addKeys(idService identity.Service) error {
+	err := idService.AddKeyFromConfig(identity.KeyPurposeP2P)
+	if err != nil {
+		panic(err)
+	}
+	err = idService.AddKeyFromConfig(identity.KeyPurposeSigning)
+	if err != nil {
+		panic(err)
+	}
+	err = idService.AddKeyFromConfig(identity.KeyPurposeEthMsgAuth)
+	if err != nil {
+		panic(err)
+	}
+	return nil
+}
+
+// CreateConfig creates a config file using provide parameters and the default config
+func CreateConfig(
+	targetDataDir, ethNodeURL, accountKeyPath, accountPassword, network string,
+	apiPort, p2pPort int64,
+	bootstraps []string,
+	txPoolAccess bool,
+	smartContractAddrs *config.SmartContractAddresses) error {
+
+	data := map[string]interface{}{
+		"targetDataDir":   targetDataDir,
+		"accountKeyPath":  accountKeyPath,
+		"accountPassword": accountPassword,
+		"network":         network,
+		"ethNodeURL":      ethNodeURL,
+		"bootstraps":      bootstraps,
+		"apiPort":         apiPort,
+		"p2pPort":         p2pPort,
+		"txpoolaccess":    txPoolAccess,
+	}
+	if smartContractAddrs != nil {
+		data["smartContractAddresses"] = smartContractAddrs
+	}
+	v, err := config.CreateConfigFile(data)
+	if err != nil {
+		return err
+	}
+	log.Infof("Config File Created: %s\n", v.ConfigFileUsed())
+	ctx, canc, _ := CommandBootstrap(v.ConfigFileUsed())
+	cfg := ctx[config.BootstrappedConfig].(*config.Configuration)
+	generateKeys(cfg)
+
+	idService := ctx[identity.BootstrappedIDService].(identity.Service)
+	id, err := createIdentity(idService)
+	if err != nil {
+		return err
+	}
+	v.Set("identityId", id.String())
+	err = v.WriteConfig()
+	if err != nil {
+		return err
+	}
+	cfg.Set("identityId", id.String())
+	log.Infof("Identity created [%s] [%x]", id.String(), id)
+	err = addKeys(idService)
+	if err != nil {
+		return err
+	}
+	canc()
+	db := ctx[storage.BootstrappedLevelDB].(*leveldb.DB)
+	db.Close()
+	return nil
+}
+
+// RunBootstrap bootstraps the node for running
+func RunBootstrap(cfgFile string) {
```

Comment:
I think we could move the Bootstraps func in another package.

For me the`cmd` package should only take care of parsing commands etc. The actual execution logic/startup logic of a command should be in another place.

Reply:
There is a constraint on where we can put this because, it can create cyclic dependancies very easily while creating many packages just because we like to see some functions in it seems kind of too much for my taste, but a pr is welcome if you find a suitable way to change this. Right now I'm just moving things so that I can use them in the `testworld`.